### PR TITLE
Apps

### PR DESCRIPTION
**Pull request by jburel**
Review the creation of application section

see https://trello.com/c/7VkRuZp9/311-web-apps-doc-review

cc @hflynn

### CODE REVIEW

File: omero/developers/Web/CreateApp.txt [hunk 1]
```diff
@@ -9,6 +9,8 @@ database mapping, since all data comes from the OMERO server and is
 saved back there. You will notice that the models.py files in each app
 are empty.
 
+All OMERO applications can now be installed from :pypi:`Python Package Index <>`.
```
Comment (joshmoore) [inline omero/developers/Web/CreateApp.txt]: Perhaps drop the "now" since this page isn't dated, and do we want to say something like "core" or "official"? i.e. all the ones that really want someone to use will by on PyPI though there may be others out there.

File: omero/developers/Web/CreateApp.txt [hunk 2]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
--  Login to your GitHub account homepage (e.g.
-   `<https://github.com/>`_\ <your-name>/) and click
+-  Login to your GitHub account homepage e.g.
+   `<https://github.com/>`_\<your-name>/) and click
    "New repository"
--  Enter the name of <your-app>, add a description and choose to add a
-   README.
 
--  Check out your new repository (into a new directory)
+-  Enter the name of your repository i.e. ``<organisation-app>`` e.g. ``omero-webtest``,
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: i.e. followed by e.g. here is a bit confusing, I'd remove one or the other, I don't think you need both

File: omero/developers/Web/CreateApp.txt [hunk 3]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
+   add a description and choose to add a ``README.rst``. Change the extension to ``.rst`` so
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: Changing to .rst reads a bit strangely given you say add 'README.rst', maybe rephrase to
"...add a README.rst (using the .rst extension allows it to be rendered correctly on PyPI)."

File: omero/developers/Web/CreateApp.txt [hunk 4]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
+   it can be rendered correctly on PyPI. Its contain should cover the goal of the app
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: "Its contain" -> "Its contents"

File: omero/developers/Web/CreateApp.txt [hunk 5]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
+   and the configuration instructions.
+
+-  Check out your new repository into a new directory
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: You can put the `::` at the end of this line to cut down the whitespace here

File: omero/developers/Web/CreateApp.txt [hunk 6]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
-       $ git clone git@github.com:<your-name>/<your-app>.git
+       $ git clone git@github.com:<your-name>/<organisation-app>.git
 
 
 -  OR: If you have not used git to create your app directory above, then
 
    ::
 
-        $ mkdir <your-app>
+        $ mkdir <organisation-app>
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: Same as above, move the `::` to the end of the previous line so cut down whitespace

File: omero/developers/Web/CreateApp.txt [hunk 7]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
 
--  In either case, you should now have a directory called ``your-app`` within
+-  In either case, you should now have a directory called ``organisation-app`` within
    a directory that is on your :envvar:`PYTHONPATH`.
 
+Make your app installable from PyPI
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is not required but it is highly recommended to make your app
+installable from PyPI. Few files need to be added:
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: "**A** few files..."

File: omero/developers/Web/CreateApp.txt [hunk 8]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
+
+- A ``MANIFEST.in`` file. This is needed in certain cases to package files not automatically included
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: I think these would read better if you formatted them as
"``filename.ext`` - a [file type] that is..."

File: omero/developers/Web/CreateApp.txt [hunk 9]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
+
+See `Packaging and Distributing Projects <https://packaging.python.org/distributing/>`_ for more details. 
 
 Add the essential files to your app
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
--  Create an empty file ``<your-app>/__init__.py`` (NB: both double
-   underscores)
--  Create :file:`urls.py`
+-  Create a directory ``<organisation_appname>`` where to add the essential files
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: "where to add the essential files" -> "to add the essential files to"

Also, I'd make the "Note the underscore" in brackets at the end of the sentence rather than adding it as a new sentence

File: omero/developers/Web/CreateApp.txt [hunk 10]
```diff
@@ -67,49 +69,70 @@ OR you could simply choose an existing location:
...
 Add the essential files to your app
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
--  Create an empty file ``<your-app>/__init__.py`` (NB: both double
-   underscores)
--  Create :file:`urls.py`
+-  Create a directory ``<organisation_appname>`` where to add the essential files 
+   e.g. ``omero_webtest``. Note the underscore
+
+-  Create an empty file ``<organisation_appname>/__init__.py``. Note double
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: again, make the note part of the sentence

File: omero/developers/Web/CreateApp.txt [hunk 11]
```diff
@@ -121,6 +144,17 @@ Add the essential files to your app
             """
             return HttpResponse("Welcome to your app home-page!")
 
+-  Create :file:`<organisation_appname>/apps.py`
+
+   ::
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: This can be moved to the end of the line of test to save all this extra whitespace

**Review by hflynn — CHANGES_REQUESTED** [review]
A bunch of pretty minor wording and formatting nitpicks, otherwise looking good

File: omero/developers/Web/CreateApp.txt [hunk 12]
```diff
@@ -52,9 +54,9 @@ We usually create a new container for a new app, and add it to the
 
 ::
 
-    $ mkdir PARENT-APP-DIR
-    $ cd PARENT-APP-DIR
-    $ export PYTHONPATH=$PYTHONPATH:/path/to/PARENT-APP-DIR
+    $ mkdir organisation-appname e.g. omero-webtest
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: I'd remove the 'e.g. omero-webtest' here

File: omero/developers/Web/CreateApp.txt [hunk 13]
```diff
@@ -67,51 +69,62 @@ OR you could simply choose an existing location:
...
 
+       $ git clone git@github.com:<your-name>/<organisation-appname>.git
 
--  OR: If you have not used git to create your app directory above, then
 
-   ::
+-  OR: If you have not used git to create your app directory above, then::
 
-        $ mkdir <your-app>
+        $ mkdir <organisation-app>
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: this is 'organisation-appname' above

File: omero/developers/Web/CreateApp.txt [hunk 14]
```diff
@@ -67,51 +69,62 @@ OR you could simply choose an existing location:
...
+This is not required but it is highly recommended to make your app
+installable from PyPI. If you opt to do so, a few files need to be added:
+
+- ``MANIFEST.in`` - a file is needed in certain cases to package files not automatically included
+
+- ``setup.cfg`` - a configuration file that contains option defaults for ``setup.py`` commands
+
+- ``setup.py`` - a set-up file used to configure various aspects of the app and also used as a command line interface for packaging the app.
+
+
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: unneeded white space

File: omero/developers/Web/CreateApp.txt [hunk 15]
```diff
@@ -67,51 +69,62 @@ OR you could simply choose an existing location:
...
+See `Packaging and Distributing Projects <https://packaging.python.org/distributing/>`_ for more details. 
 
 Add the essential files to your app
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
--  Create an empty file ``<your-app>/__init__.py`` (NB: both double
-   underscores)
--  Create :file:`urls.py`
+-  Create a directory ``<organisation_appname>`` (note the underscore) to add the essential files to
+   e.g. ``omero_webtest``
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: I would move the 'e.g. omero-webtest' to earlier in the sentence:
"Create a directory ``<organisation_appname>`` e.g. ``omero_webtest`` (note the underscore) to add the essential files to

File: omero/developers/Web/CreateApp.txt [hunk 16]
```diff
@@ -121,6 +134,15 @@ Add the essential files to your app
...
 
+-  Create :file:`<organisation_appname>/apps.py`::
+
+        from django.apps import AppConfig
+
+        class AppNameAppConfig(AppConfig):
+            name = "organisation_appname"
+            label = "appname"
+
+
```
Comment (hflynn) [inline omero/developers/Web/CreateApp.txt]: unneeded extra blank line here

**Comment by hflynn** [comment]
A few more nitpicks and @mtbc would probably say we should be spelling 'organisation' as 'organization' throughout ;)

File: omero/developers/Web/LinkingFromWebclient.txt [hunk 17]
```diff
@@ -156,7 +156,7 @@ For more control over the enabled status of your plugin or to configure how urls
 are created from selected objects, you can write JavaScript functions
 that handle these steps.
 These functions use the label specified above as an ID for your ``Open with``
-option. In this example it is "xyz_viewer".
+option. In this example it is ``xyz_viewe``.
```
Comment (mtbc) [inline omero/developers/Web/LinkingFromWebclient.txt]: xyz_viewer

**Comment by jburel** [comment]
All the requested changes have now been done

File: omero/developers/Web/LinkingFromWebclient.txt [hunk 18]
```diff
@@ -100,17 +100,17 @@ For example:
 
     $ bin/omero config append omero.web.open_with '["xyz_viewer", "url_name", {"supported_objects": ["image"]}]'
 
-This will create a menu option named "xyz_viewer" that is only enabled when a
+This will create a menu option named``xyz_viewe` that is only enabled when a
```
Comment (hflynn) [inline omero/developers/Web/LinkingFromWebclient.txt]: Erm, shouldn't this be viewer?

File: omero/developers/Web/LinkingFromWebclient.txt [hunk 19]
```diff
@@ -100,17 +100,17 @@ For example:
...
 single "image" is selected.
 
-The unique ID string, "xyz_viewer" can be used to identify your plugin
+The unique ID string,``xyz_viewe`` can be used to identify your plugin
```
Comment (hflynn) [inline omero/developers/Web/LinkingFromWebclient.txt]: As above, you seem to have introduced a typo here

File: omero/developers/Web/WebclientPlugin.txt [hunk 20]
```diff
@@ -72,22 +71,22 @@ Choose an element ID
...
-/templates/<app-name>/webclient_plugins/right_plugin_rois.html.
+`/templates/<organization_appname>/` directory and named `.html`, e.g.
+/templates/<organization_appname>/webclient_plugins/right_plugin_rois.html.
 All the JavaScript should be within `<script>` and `</script>` tags.
 Your plugin initialization should happen after the page has loaded, so you use
 the jQuery on-ready function.
 
-You use custom jQuery functions, called 'omeroweb_right_plugin' or
-'omeroweb_center_plugin', to initialize the webclient plugin. These will
+You use custom jQuery functions, called `omeroweb_right_plugin` or
```
Comment (hflynn) [inline omero/developers/Web/WebclientPlugin.txt]: Double backticks here and below would be better

**Comment by hflynn** [comment]
Just a couple more minor nitpicks, otherwise this is looking good now thanks.

**Comment by jburel** [comment]
@hflynn fixed

**Review by hflynn — APPROVED** [review]
